src/main/java/seedu/mark/model/ModelManager.java

```diff
@@ -184,7 +204,8 @@ public boolean equals(Object obj) {
         ModelManager other = (ModelManager) obj;
         return versionedMark.equals(other.versionedMark)
                 && userPrefs.equals(other.userPrefs)
-                && filteredBookmarks.equals(other.filteredBookmarks);
+                && filteredBookmarks.equals(other.filteredBookmarks)
+                && ((currentUrl.getValue() == null && other.currentUrl.getValue() == null)
```

Comment:
this will throw NPE for currentUrl.getValue() == null and other.currentUrl.getValue != null:

add this test to ModelManagerTest.java

```
// this url null, other url nonnull -> error
        modelManager = new ModelManager(mark, userPrefs);
        ModelManager modelManagerCopy = new ModelManager(mark, userPrefs);
        modelManagerCopy.setCurrentUrl(new Url("http://google.com"));
        modelManager.setCurrentUrl(null);
        assertTrue(modelManager.equals(modelManagerCopy));
```

![image](https://user-images.githubusercontent.com/3646725/66701847-3cc26a80-ed33-11e9-8865-1df85ea85660.png)

Reply:
You're right! Thanks